src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -169,38 +170,54 @@ private Collector getHybridQueryCollector() {
      * @return search results that can be reduced be the caller
      */
     @Override
-    public ReduceableSearchResult reduce(Collection<Collector> collectors) {
+    public ReduceableSearchResult reduce(Collection<Collector> collectors) throws IOException {
         final List<HybridSearchCollector> hybridSearchCollectors = getHybridSearchCollectors(collectors);
         if (hybridSearchCollectors.isEmpty()) {
             throw new IllegalStateException("cannot collect results of hybrid search query, there are no proper collectors");
         }
         return reduceSearchResults(getSearchResults(hybridSearchCollectors));
     }
 
-    private List<ReduceableSearchResult> getSearchResults(final List<HybridSearchCollector> hybridSearchCollectors) {
+    private List<ReduceableSearchResult> getSearchResults(final List<HybridSearchCollector> hybridSearchCollectors) throws IOException {
         List<ReduceableSearchResult> results = new ArrayList<>();
         DocValueFormat[] docValueFormats = getSortValueFormats(sortAndFormats);
         for (HybridSearchCollector collector : hybridSearchCollectors) {
             boolean isSortEnabled = docValueFormats != null;
-            TopDocsAndMaxScore topDocsAndMaxScore = getTopDocsAndAndMaxScore(collector, isSortEnabled);
-            results.add((QuerySearchResult result) -> reduceCollectorResults(result, topDocsAndMaxScore, docValueFormats));
+            boolean isCollapseEnabled = collector instanceof HybridCollapsingTopDocsCollector;
+            TopDocsAndMaxScore topDocsAndMaxScore = getTopDocsAndAndMaxScore(collector, isSortEnabled, isCollapseEnabled);
+            results.add(
+                (QuerySearchResult result) -> reduceCollectorResults(
+                    result,
+                    topDocsAndMaxScore,
+                    isCollapseEnabled && isSortEnabled == false ? new DocValueFormat[] { DocValueFormat.RAW } : docValueFormats
+                )
+            );
         }
         return results;
     }
 
-    private TopDocsAndMaxScore getTopDocsAndAndMaxScore(final HybridSearchCollector hybridSearchCollector, final boolean isSortEnabled) {
+    private TopDocsAndMaxScore getTopDocsAndAndMaxScore(
+        final HybridSearchCollector hybridSearchCollector,
+        final boolean isSortEnabled,
+        final boolean isCollapseEnabled
+    ) throws IOException {
         List topDocs = hybridSearchCollector.topDocs();
-        if (isSortEnabled) {
-            return getSortedTopDocsAndMaxScore(topDocs, hybridSearchCollector);
+        if (isSortEnabled || isCollapseEnabled) {
```

Comment:
Can sort and collapse work together? Then what would happen in that situation?

Reply:
Yes, they work together and in that case the collapse code would be followed with the sort passed into it.  The only reason for checking both is the case where only one of them is enabled.  If collapse is enabled without sort, we still need to create FieldDocs instead of ScoreDocs because CollapseTopFieldDocs uses them.